src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DataAnnotationsMetadataProvider.cs

```diff
@@ -84,7 +84,7 @@ public void GetDisplayMetadata([NotNull] DisplayMetadataProviderContext context)
             // DisplayName
             if (displayAttribute != null)
             {
-                displayMetadata.DisplayName = displayAttribute.GetName();
+                displayMetadata.DisplayName = (propertyName) => displayAttribute.GetName();
```

Comment:
Typically when you want to be careful with a capturing lambda is on a known hot-path or in a tight loop. The issue that is a capturing lambda is an allocation.

Three reasons not to care:
- Measure first, then optimize
- This is definitely not a hot path (it's already cached)
- What non-allocating option would you choose?